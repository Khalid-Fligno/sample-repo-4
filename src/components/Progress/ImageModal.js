import React from 'react';
import PropTypes from 'prop-types';
import {
  View,
  StyleSheet,
  TouchableOpacity,
  Text,
  Dimensions,
  ScrollView,
} from 'react-native';
import Modal from 'react-native-modal';
import FastImage from 'react-native-fast-image';
import fonts from '../../styles/fonts';
import colors from '../../styles/colors';

const { width } = Dimensions.get('window');

export default class ImageModal extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
    };
  }
  render() {
    const {
      imageModalVisible,
      toggleImageModal,
      color,
      imageSource,
    } = this.props;
    return (
      <Modal
        isVisible={imageModalVisible}
        animationIn="fadeIn"
        animationInTiming={400}
        animationOut="fadeOut"
        animationOutTiming={600}
      >
        <View style={styles.imageModalContainer}>
          <View style={styles.imageModalImageContainer}>
            <ScrollView
              bounces={false}
              bouncesZoom={false}
              maximumZoomScale={2}
              minimumZoomScale={1}
            >
              <FastImage
                style={styles.imageStyle}
                source={imageSource}
              />
            </ScrollView>
          </View>
          <View style={styles.imageModalButtonContainer}>
            <TouchableOpacity
              title="DONE"
              onPress={toggleImageModal}
              style={[
                styles.modalButton,
                color === 'coral' && styles.modalButtonCoral,
                color === 'violet' && styles.modalButtonViolet,
                color === 'blue' && styles.modalButtonBlue,
                color === 'green' && styles.modalButtonGreen,
                color === 'red' && styles.modalButtonRed,
              ]}
            >
              <Text style={styles.modalButtonText}>
                DONE
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </Modal>
    );
  }
}

ImageModal.propTypes = {
  imageModalVisible: PropTypes.bool.isRequired,
  toggleImageModal: PropTypes.func.isRequired,
  color: PropTypes.string,
  imageSource: PropTypes.objectOf(PropTypes.string).isRequired,
};

ImageModal.defaultProps = {
  color: undefined,
};

const styles = StyleSheet.create({
  imageModalContainer: {
    flexShrink: 1,
    alignItems: 'center',
    justifyContent: 'space-between',
    borderRadius: 4,
    overflow: 'hidden',
  },
  imageModalImageContainer: {
    width: '100%',
    backgroundColor: colors.grey.standard,
  },
  imageStyle: {
    width: width - 30,
    height: (width - 30) * 1.33,
  },
  imageModalButtonContainer: {
    backgroundColor: colors.white,
    width: '100%',
  },
  modalButton: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.charcoal.standard,
    height: 50,
    width: '100%',
  },
  modalButtonCoral: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.themeColor.color,
    height: 50,
    width: '100%',
  },
  modalButtonViolet: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.violet.standard,
    height: 50,
    width: '100%',
  },
  modalButtonBlue: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.blue.standard,
    height: 50,
    width: '100%',
  },
  modalButtonGreen: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.green.standard,
    height: 50,
    width: '100%',
  },
  modalButtonRed: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.red.standard,
    height: 50,
    width: '100%',
  },
  modalButtonText: {
    fontFamily: fonts.bold,
    fontSize: 14,
    color: colors.white,
    marginTop: 3,
  },
});
