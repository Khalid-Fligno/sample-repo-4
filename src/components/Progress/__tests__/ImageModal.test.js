import React from 'react';
import renderer from 'react-test-renderer';
import ImageModal from '../ImageModal';

describe('Image Modal', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ImageModal
      imageModalVisible
      toggleImageModal={jest.fn()}
      color="Test"
      imageSource={{ uri: 'test' }}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
