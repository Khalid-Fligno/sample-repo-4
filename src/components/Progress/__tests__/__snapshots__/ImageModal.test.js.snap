// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Modal renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgb(188, 188, 189)",
            "width": "100%",
          }
        }
      >
        <RCTScrollView
          bounces={false}
          bouncesZoom={false}
          maximumZoomScale={2}
          minimumZoomScale={1}
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "height": 957.6,
                    "width": 720,
                  },
                ]
              }
            >
              <FastImageView
                resizeMode="cover"
                source={
                  Object {
                    "uri": "test",
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgb(255, 255, 255)",
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(38, 38, 40)",
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(255, 255, 255)",
                "fontFamily": "GothamBold",
                "fontSize": 14,
                "marginTop": 3,
              }
            }
          >
            DONE
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
